Citations

### RAST
If you use the results of this annotation in your work, please cite:
The RAST Server: Rapid Annotations using Subsystems Technology.
Aziz RK, Bartels D, Best AA, DeJongh M, Disz T, Edwards RA, Formsma K, Gerdes S, Glass EM, Kubal M, Meyer F, Olsen GJ, Olson R, Osterman AL, Overbeek RA, McNeil LK, Paarmann D, Paczian T, Parrello B, Pusch GD, Reich C, Stevens R, Vassieva O, Vonstein V, Wilke A, Zagnitko O.
BMC Genomics, 2008, [ PubMed entry ]
The SEED and the Rapid Annotation of microbial genomes using Subsystems Technology (RAST).
Overbeek R, Olson R, Pusch GD, Olsen GJ, Davis JJ, Disz T, Edwards RA, Gerdes S, Parrello B, Shukla M, Vonstein V, Wattam AR, Xia F, Stevens R.
Nucleic Acids Res. 2014 [ PubMed entry ]
RASTtk: A modular and extensible implementation of the RAST algorithm for building custom annotation pipelines and annotating batches of genomes.
Brettin T, Davis JJ, Disz T, Edwards RA, Gerdes S, Olsen GJ, Olson R, Overbeek R, Parrello B, Pusch GD, Shukla M, Thomason JA, Stevens R, Vonstein V, Wattam AR, Xia F.
Sci Rep., 2015, [ PubMed entry ]

### Prokka
Seemann T, "Prokka: Rapid Prokaryotic Genome Annotation",
  Bioinformatics, 2014 Jul 15;30(14):2068-9.

  PMID:24642063
  doi:10.1093/bioinformatics/btu153
  http://www.ncbi.nlm.nih.gov/pubmed/24642063

### Blast

Gene [Internet]. Bethesda (MD): National Library of Medicine (US), National Center for Biotechnology Information; 2004 â€“ [cited YYYY Mmm DD]. Available from: https://www.ncbi.nlm.nih.gov/gen

### ClustAGE

Ozer EA. ClustAGE: a tool for clustering and distribution analysis of bacterial accessory genomic elements. BMC Bioinformatics 2018 19:150

### Spine/ AGEnt

Ozer EA, Allen JP, and Hauser AR. Characterization of the core and accessory genomes of Pseudomonas aeruginosa using bioinformatic tools Spine and AGEnt. BMC Genomics 2014 15:737

### Paper for phylogeny of Salmonella

Worley, J., Meng, J., Allard, M.W., Brown, E.W. and Timme, R.E., 2018. Salmonella enterica Phylogeny Based on Whole-Genome Sequencing Reveals Two New Clades and Novel Patterns of Horizontally Acquired Genetic Elements. MBio, 9(6), pp.e02303-18.

### Paper for generalist vs host adapted graphic

Feasey, N.A., Dougan, G., Kingsley, R.A., Heyderman, R.S. and Gordon, M.A., 2012. Invasive non-typhoidal salmonella disease: an emerging and neglected tropical disease in Africa. The Lancet, 379(9835), pp.2489-2499.

### Papers for strain Information

Fookes, M., Schroeder, G.N., Langridge, G.C., Blondel, C.J., Mammina, C., Connor, T.R., Seth-Smith, H., Vernikos, G.S., Robinson, K.S., Sanders, M. and Petty, N.K., 2011. Salmonella bongori provides insights into the evolution of the Salmonellae. PLoS pathogens, 7(8), p.e1002191.

Fricke, W.F., Mammel, M.K., McDermott, P.F., Tartera, C., White, D.G., LeClerc, J.E., Ravel, J. and Cebula, T.A., 2011. Comparative genomics of 28 Salmonella enterica isolates: evidence for CRISPR-mediated adaptive sublineage evolution. Journal of bacteriology, 193(14), pp.3556-3568.

Kingsley, R.A., Kay, S., Connor, T., Barquist, L., Sait, L., Holt, K.E., Sivaraman, K., Wileman, T., Goulding, D., Clare, S. and Hale, C., 2013. Genome and transcriptome adaptation accompanying emergence of the definitive type 2 host-restricted Salmonella enterica serovar Typhimurium pathovar. MBio, 4(5), pp.e00565-13.
