# Final Project

## Introduction

This is a final project for the [Comparative Genomics](https://github.com/Yale-EEB723/syllabus) seminar in the spring of 2019. This project will analyze publicly available RNA-seq data obtained from the SRA database for presence and absence of important immune genes, the major histocompatibility complex (MHC) genes, in a clade of teleost fishes known as notothenioids.  

## Background

The major histocompatibility complex (MHC) represents a fundamental component of the vertebrate adaptive immune system, and its functionailites and pathways have long been thought to be highly conserved across species. However, significant modifications of the adaptive immune system have been noted, and in some cases have been associated with shifts in speciation rate in those lineages.

## The goal

The primary goal of this work is to investigate the evolution of MHCI and MHCII lineages within and among notothenioid species. 

## The data

Previously-sequenced and publicly-available transcriptomes representing 13 Antarctic and non-Antarctic notothenioid species were obtained from the NCBI Sequence Read Archive.

This project will also require a reference genome. For this, we will use the genome of the Antarctic bullhead notothen (Notothenia coriiceps), which will be obtained from BioProject. 

## Methods

To investigate MHC gene evolution in the Antarctic notothenioids, we compiled an RNA-seq dataset including 13 Antarctic and non-Antarctic notothenioid species and 9 different tissues (see Table 1). Immune-related genes will be identified from the notothenioid transcripts using a combination of BLAST and HMMER searches. We will then infer the phylogenetic relationships of the sequences identified as MHC genes and we will estimate relative rates of evolution for sequences that are highly conserved or diverged between notothenioid species. We will pay special attention to comparisons between genes in Antarctic notothenioids and genes in non-Antarctic notothenioids to determine genes that may be unique to Anatarctic notothenioids. 

## Results


## Assessment

Was it successful in achieving the initial goal?

What are the main obstacles encountered?

What would you have done differently?

What are future directions this could go in?

## References

Chen, Zuozhou, C-H. Christina Cheng, Junfang Zhang, Lixue Cao, Lei Chen, Longhai Zhou, Yudong Jin et al. "Transcriptomic and genomic evolution under constant cold in Antarctic notothenioid fish." Proceedings of the National Academy of Sciences 105, no. 35 (2008): 12944-12949.

Kim, Bo-Mi, Angel Amores, Seunghyun Kang, Do-Hwan Ahn, Jin-Hyoung Kim, Il-Chan Kim, Jun Hyuck Lee et al. "Antarctic blackfin icefish genome reveals adaptations to extreme environments." Nature ecology & evolution 3, no. 3 (2019): 469.

Malmstrøm, Martin, Michael Matschiner, Ole K. Tørresen, Bastiaan Star, Lars G. Snipen, Thomas F. Hansen, Helle T. Baalsrud et al. "Evolution of the immune system influences speciation rates in teleost fishes." Nature Genetics 48, no. 10 (2016): 1204.