# Final Project

## Introduction

This is a final project for the [Comparative Genomics](https://github.com/Yale-EEB723/syllabus) seminar in the spring of 2019. This project used RNA-seq data published on SRA to test the validity of the group Cyclostomata, the living jawless fishes.

## The goal

Molecular evidence supporting Cyclostoma has thus far been limited to a few mitochondrial genes. The goal of this preliminary study is to use transcriptome data to infer a tree that will test the phylogenetic relationships between hagfishes (Myxiniformes), lampreys (Petromyzontiformes), and jawed vertebrates (Gnathostomes).

## The data

Paired-end sequences are accessible on SRA for the following lamprey species:

Petromyzontinae: Petromyzon marinus, Ichthyomyzon castaneus, Ichthyomyzon fossor

Lampetrinae: Lampetra planeri, Lethenteron camtschaticum, and Lethenteron reissneri,

Geotriidae: Geotria australis

and the two hagfish species (Eptatretus burgeri and Eptatretus cirrhatus).

I have yet to determine which tunicate and lancelet representatives would be best (or possible) to include in this study, as well as which and how many representatives of the jawed vertebrates would be most useful to include. 

## Background

Motivation for the project....

How it fits in with other work...

What the reader needs to know to understand the project


## Methods
I plan to use the Agalma pipelines to identify homologous sequences across the multiple species, create nucleotide alignments for each set of homologus sequence, and infer a preliminary species tree with RAxML.

## Results


## Assessment

Was it successful in achieving the initial goal?

What are the main obstacles encountered?

What would you have done differently?

What are future directions this could go in?

## References
